### PR TITLE
Some testing robustness improvement and opentracing error case

### PR DESCRIPTION
**Pull request by juangon**
@Ladicek @penehyba I added some improvements to check "key" fields too, and added test case for error responses tracing too.

### CODE REVIEW

File: microprofile/microprofile-opentracing-1.3/src/main/java/org/wildfly/swarm/ts/microprofile/opentracing/v13/TracedService.java [hunk 1]
```diff
@@ -63,4 +63,12 @@ public String twoWildcard(long id, String txt) {
         tracer.activeSpan().log("twoWildcard: " + id + ", " + txt);
         return "Hello from twoWildcard: " + id + ", " + txt;
     }
+
+    @SuppressWarnings("null")
+    @Traced
+    public String traceError() {
+        tracer.activeSpan().log("traceError");
+        String valueReturn = null;
+        return valueReturn.toString();
```
Comment (Ladicek) [inline microprofile/microprofile-opentracing-1.3/src/main/java/org/wildfly/swarm/ts/microprofile/opentracing/v13/TracedService.java]: How about just `throw new RuntimeException();`?

File: microprofile/microprofile-opentracing-1.3/src/test/java/org/wildfly/swarm/ts/microprofile/opentracing/v13/MPOpenTracing13Test.java [hunk 2]
```diff
@@ -106,16 +124,68 @@ public void trace() {
 
             assertThat(data.size()).isEqualTo(EXPECTED_TRACES.size());
 
-            EXPECTED_TRACES.forEach(trace -> checkTrace(data,
-                                                        trace.getOperationName(),
-                                                        trace.getLog(),
-                                                        trace.getChildOperationName(),
-                                                        trace.getChildLog()));
+            EXPECTED_TRACES.forEach(trace -> { 
+                if (!trace.getTag().containsKey("error")) { //Only non error traces
```
Comment (Ladicek) [inline microprofile/microprofile-opentracing-1.3/src/test/java/org/wildfly/swarm/ts/microprofile/opentracing/v13/MPOpenTracing13Test.java]: Create a `stream()` and `filter()` out the error ones?

**Comment by juangon** [comment]
@Ladicek added your suggestions, thanks for those

**Comment by penehyba** [comment]
LGTM, good job!

**Comment by Ladicek** [comment]
I did a bit of a refactoring to remove unnecessary complexity, and merged. Thanks.

**Pull request merged**